cpp/include/raft/core/resource/cuda_stream.hpp

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cuda_runtime.h>
+#include <raft/core/interruptible.hpp>
+#include <raft/core/resource/resource_types.hpp>
+#include <raft/core/resources.hpp>
+#include <raft/util/cudart_utils.hpp>
+#include <rmm/cuda_stream_view.hpp>
+
+namespace raft::resource {
+class cuda_stream_resource : public resource {
+ public:
+  cuda_stream_resource(rmm::cuda_stream_view stream_view = rmm::cuda_stream_per_thread)
+    : stream(stream_view)
+  {
+  }
+  void* get_resource() override { return &stream; }
+
+  ~cuda_stream_resource() override {}
+
+ private:
+  rmm::cuda_stream_view stream;
+};
+
+/**
+ * Factory that knows how to construct a
+ * specific raft::resource to populate
+ * the res_t.
+ */
+class cuda_stream_resource_factory : public resource_factory {
+ public:
+  cuda_stream_resource_factory(rmm::cuda_stream_view stream_view = rmm::cuda_stream_per_thread)
+    : stream(stream_view)
+  {
+  }
+  resource_type get_resource_type() override { return resource_type::CUDA_STREAM_VIEW; }
+  resource* make_resource() override { return new cuda_stream_resource(stream); }
+
+ private:
+  rmm::cuda_stream_view stream;
+};
+
+/**
+ * Load a cudaStream_t from a res (and populate it on the res
```

Comment:
Is the naming `cudaStream_t` intentional, or should it be `cuda_stream_view` here?